tmt/steps/report/junit.py

```diff
@@ -58,40 +69,188 @@ def duration_to_seconds(duration: Optional[str]) -> Optional[int]:
             f"Malformed duration '{duration}' ({error}).")
 
 
+class ResultsContext:
+    """
+    A class which keeps the results context (especially the result summary) for
+    JUnit template.
+    """
+
+    def __init__(self, results: list[tmt.Result]) -> None:
+        self._results = results
+
+    def __iter__(self) -> Iterator[tmt.Result]:
+        """ Possibility to iterate over results by iterating an instance """
+        return iter(self._results)
+
+    def __len__(self) -> int:
+        """ Returns the number of results """
+        return len(self._results)
+
+    @functools.cached_property
+    def executed(self) -> list[tmt.Result]:
+        """ Returns results of all executed tests """
+        return [r for r in self._results if r.result != ResultOutcome.INFO]
+
+    @functools.cached_property
+    def skipped(self) -> list[tmt.Result]:
+        """ Returns results of skipped tests """
+        return [r for r in self._results if r.result == ResultOutcome.INFO]
+
+    @functools.cached_property
+    def failed(self) -> list[tmt.Result]:
+        """ Returns results of failed tests """
+        return [r for r in self._results if r.result == ResultOutcome.FAIL]
+
+    @functools.cached_property
+    def errored(self) -> list[tmt.Result]:
+        """ Returns results of tests with error/warn outcome """
+        return [r for r in self._results if r.result in [
+                ResultOutcome.ERROR,
+                ResultOutcome.WARN]]
+
+    @functools.cached_property
+    def duration(self) -> int:
+        """ Returns the total duration of all tests in seconds """
+        return sum([duration_to_seconds(r.duration) or 0 for r in self._results])
+
+
 def make_junit_xml(
-        report: 'ReportPlugin[ReportJUnitData]|ReportPlugin[ReportPolarionData]'
-        ) -> 'junit_xml.TestSuite':
-    """ Create junit xml object """
-    junit_xml = import_junit_xml()
+        phase: 'ReportPlugin[ReportJUnitData]|ReportPlugin[ReportPolarionData]',
```

Comment:
I'm afraid both are correct :) It's a plugin *instance*, and represents one *phase* of the `report` step.

One note though: `ReportPlugin[ReportStepData]` or even `ReportPlugin[Any]` would probably be better, because this list will never be complete, anyone can create an out-of-the-tree plugin & use `junit` plugin to render its templates. Limiting it to just two examples does not seem correct.

Reply:
Ok, I've changed it to `ReportPlugin[Any]`. I was keeping it the same way as it was before.